### ISSUE TITLE
False positives regarding old limitations of Busybox sh (Alpine)

### ISSUE THREAD

**Issue by reitzig** [hadolint/hadolint#1084]
From what I can tell, Busybox defaults to `ash` these days, which doesn't have the same limitations as its `sh` used to have. That seems to have made some rules obsolete, that is they are false positives now.

For example, I'm getting warnings from these:

- DL4006 -- but
  ```dockerfile
  SHELL ["/bin/sh", "-e", "-u", "-o", "pipefail", "-c"]
  ```
  builds and runs fine
- SC3037 -- but:
  ```
  ❯ docker run --rm -it alpine:3.21 sh -c 'echo -e "abc\ndef"'
  abc
  def
  ```

**Comment by kiler129** [comment]
I would argue it's *not* a false positive in this case. In POSIX `sh` the `-e` undefined, making the SC warning correct. In bare `ash` the `-e` flag is also undefined.

The best of both worlds would probably be using:
```dockerfile
FROM alpine:3.21
SHELL ["/bin/ash", "-euxo", "pipefail", "-c"]
RUN printf "Hello\tWorld\n"
```

**Comment by reitzig** [comment]
Maybe [the message](https://github.com/hadolint/hadolint/blob/28fc39f92e2768d8cc9f94d6e2fc16bd4397f092/src/Hadolint/Rule/DL4006.hs#L14-L17) is misleading:

> Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check

This seems to be tied directly to Alpine and busybox, to which it doesn't apply anymore. If the rule is intended to discourage using non-POSIX commands without explicit `SHELL` in general, I suggest it be rephrased.
